### PR TITLE
Housekeeping: Explicitly encode QStrings to utf8.

### PR DESCRIPTION
**Pull request by daleglass**
Fixes some warnings. Shouldn't make any functional difference.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-_28add85c.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-.tar_28add85c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-_28add85c.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-.tar_28add85c.gz

**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-Interface-PR1485-0711a3f-_28add85c.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-_28add85c.exe

**build (self-hosted_debian-11_aarch64, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-.tar_28add85c.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-.tar_28add85c.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-0711a3f-_28add85c.sh

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-.tar_8c60ba65.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-_8c60ba65.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-.tar_8c60ba65.Z

**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-_8c60ba65.dmg

**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-Interface-PR1485-fabed6b-_8c60ba65.dmg

**build (self-hosted_debian-11_aarch64, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-.tar_8c60ba65.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-.tar_8c60ba65.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-_8c60ba65.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-fabed6b-_8c60ba65.exe

File: libraries/networking/src/AccountManager.cpp [hunk 1]
```diff
@@ -569,7 +569,7 @@ void AccountManager::requestAccessToken(const QString& login, const QString& pas
     postData.append("grant_type=password&");
     postData.append("username=" + QUrl::toPercentEncoding(login) + "&");
     postData.append("password=" + QUrl::toPercentEncoding(password) + "&");
-    postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE);
+    postData.append(QString("scope=%1").arg(ACCOUNT_MANAGER_REQUESTED_SCOPE).toUtf8());
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:572]: Simpler change...
```
postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE.toUtf8());
```
No warnings on Windows.

File: libraries/networking/src/AccountManager.cpp [hunk 2]
```diff
@@ -590,10 +590,10 @@ void AccountManager::requestAccessTokenWithAuthCode(const QString& authCode, con
...
     QByteArray postData;
     postData.append("grant_type=authorization_code&");
-    postData.append("client_id=" + clientId + "&");
-    postData.append("client_secret=" + clientSecret + "&");
-    postData.append("code=" + authCode + "&");
-    postData.append("redirect_uri=" + QUrl::toPercentEncoding(redirectUri));
+    postData.append(QString("client_id=%1&").arg(clientId).toUtf8());
+    postData.append(QString("client_secret=%1&").arg(clientSecret).toUtf8());
+    postData.append(QString("code=%1&").arg(authCode).toUtf8());
+    postData.append(QByteArray("redirect_uri=") + QUrl::toPercentEncoding(redirectUri));
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:596]: Simpler...
```
    postData.append("client_id=" + clientId.toUtf8() + "&");
    postData.append("client_secret=" + clientSecret.toUtf8() + "&");
    postData.append("code=" + authCode.toUtf8() + "&");
    postData.append("redirect_uri=" + QUrl::toPercentEncoding(redirectUri));
    ```
No warnings on Windows.

File: libraries/networking/src/AccountManager.cpp [hunk 3]
```diff
@@ -613,8 +613,8 @@ void AccountManager::requestAccessTokenWithSteam(QByteArray authSessionTicket) {
 
     QByteArray postData;
     postData.append("grant_type=password&");
-    postData.append("steam_auth_ticket=" + QUrl::toPercentEncoding(authSessionTicket) + "&");
-    postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE);
+    postData.append(QByteArray("steam_auth_ticket=") + QUrl::toPercentEncoding(authSessionTicket) + "&");
+    postData.append(QString("scope=%1").arg(ACCOUNT_MANAGER_REQUESTED_SCOPE).toUtf8());
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:617]: Similarly here and elsewhere in this PR.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-Interface-PR1485-52c758c-_33bf554b.dmg

**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-_33bf554b.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-.tar_33bf554b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-_33bf554b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-.tar_33bf554b.Z

**build (self-hosted_debian-11_aarch64, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-.tar_33bf554b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-_33bf554b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-52c758c-.tar_33bf554b.Z

File: libraries/networking/src/AccountManager.cpp [hunk 4]
```diff
@@ -590,10 +590,10 @@ void AccountManager::requestAccessTokenWithAuthCode(const QString& authCode, con
...
     QByteArray postData;
     postData.append("grant_type=authorization_code&");
-    postData.append("client_id=" + clientId + "&");
-    postData.append("client_secret=" + clientSecret + "&");
-    postData.append("code=" + authCode + "&");
-    postData.append("redirect_uri=" + QUrl::toPercentEncoding(redirectUri));
+    postData.append("client_id=" + clientId.toUtf8() + "&");
+    postData.append("client_secret=" + clientSecret.toUtf8() + "&");
+    postData.append("code=" + authCode.toUtf8() + "&");
+    postData.append(QByteArray("redirect_uri=") + QUrl::toPercentEncoding(redirectUri));
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:596]:
```suggestion
    postData.append("redirect_uri=" + QUrl::toPercentEncoding(redirectUri));
```

File: libraries/networking/src/AccountManager.cpp [hunk 5]
```diff
@@ -613,8 +613,8 @@ void AccountManager::requestAccessTokenWithSteam(QByteArray authSessionTicket) {
 
     QByteArray postData;
     postData.append("grant_type=password&");
-    postData.append("steam_auth_ticket=" + QUrl::toPercentEncoding(authSessionTicket) + "&");
-    postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE);
+    postData.append(QByteArray("steam_auth_ticket=") + QUrl::toPercentEncoding(authSessionTicket) + "&");
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:616]:
```suggestion
    postData.append("steam_auth_ticket=" + QUrl::toPercentEncoding(authSessionTicket) + "&");
```

File: libraries/networking/src/AccountManager.cpp [hunk 6]
```diff
@@ -635,9 +635,9 @@ void AccountManager::requestAccessTokenWithOculus(const QString& nonce, const QS
 
     QByteArray postData;
     postData.append("grant_type=password&");
-    postData.append("oculus_nonce=" + nonce + "&");
-    postData.append("oculus_id=" + oculusID + "&");
-    postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE);
+    postData.append("oculus_nonce=" + nonce.toUtf8() + "&");
+    postData.append("oculus_id=" + oculusID.toUtf8() + "&");
+    postData.append("scope=" +  ACCOUNT_MANAGER_REQUESTED_SCOPE.toUtf8());
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:640]:
```suggestion
    postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE.toUtf8());
```

File: libraries/networking/src/AccountManager.cpp [hunk 7]
```diff
@@ -666,8 +666,8 @@ void AccountManager::refreshAccessToken() {
 
         QByteArray postData;
         postData.append("grant_type=refresh_token&");
-        postData.append("refresh_token=" + QUrl::toPercentEncoding(_accountInfo.getAccessToken().refreshToken) + "&");
-        postData.append("scope=" + ACCOUNT_MANAGER_REQUESTED_SCOPE);
+        postData.append(QByteArray("refresh_token=") + QUrl::toPercentEncoding(_accountInfo.getAccessToken().refreshToken) + "&");
```
Comment (ctrlaltdavid) [inline libraries/networking/src/AccountManager.cpp:669]:
```suggestion
        postData.append("refresh_token=" + QUrl::toPercentEncoding(_accountInfo.getAccessToken().refreshToken) + "&");
```

File: libraries/networking/src/DomainAccountManager.cpp [hunk 8]
```diff
@@ -90,9 +90,9 @@ void DomainAccountManager::requestAccessToken(const QString& username, const QSt
     // - Ignores "state" parameter.
     QByteArray formData;
     formData.append("grant_type=password&");
-    formData.append("username=" + QUrl::toPercentEncoding(username) + "&");
-    formData.append("password=" + QUrl::toPercentEncoding(password) + "&");
-    formData.append("client_id=" + _currentAuth.clientID);
+    formData.append(QByteArray("username=") + QUrl::toPercentEncoding(username) + "&");
+    formData.append(QByteArray("password=") + QUrl::toPercentEncoding(password) + "&");
+    formData.append(QString("client_id=%1").arg(_currentAuth.clientID).toUtf8());
```
Comment (ctrlaltdavid) [inline libraries/networking/src/DomainAccountManager.cpp:95]:
```suggestion
    formData.append("username=" + QUrl::toPercentEncoding(username) + "&");
    formData.append("password=" + QUrl::toPercentEncoding(password) + "&");
    formData.append("client_id=" + _currentAuth.clientID.toUtf8());
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-10.15, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-Interface-PR1485-3d4163c-_309e2b0e.dmg

**build (macOS-10.15, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-_309e2b0e.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-.tar_309e2b0e.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-.tar_309e2b0e.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-_309e2b0e.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-_309e2b0e.exe

**build (self-hosted_debian-11_aarch64, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-_309e2b0e.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-.tar_309e2b0e.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1485/Vircadia-PR1485-3d4163c-.tar_309e2b0e.gz

**Review by ctrlaltdavid — APPROVED** [review]

**Review by digisomni — APPROVED** [review]
Seems to work fine for me on Windows, tested logins on serverless, a server, and a localhost server.

**Pull request merged**